R/print_gbif.R

```diff
@@ -1,14 +1,33 @@
 #' @export
 print.gbif <- function(x, ...) {
   if (
-    if (is.null(attr(x, "type"))) FALSE else attr(x, "type") == "single" &&
-    all(c('meta', 'data', 'hierarchy', 'media') %in% names(x))
-  ) {
-    cat(rgbif_wrap(sprintf("Records found [%s]", x$meta$count)), "\n")
-    cat(rgbif_wrap(sprintf("Records returned [%s]", NROW(x$data))), "\n")
-    cat(rgbif_wrap(sprintf("No. unique hierarchies [%s]", length(x$hierarchy))), "\n")
-    cat(rgbif_wrap(sprintf("No. media records [%s]", length(x$media))), "\n")
-    cat(rgbif_wrap(sprintf("No. facets [%s]", length(x$facets))), "\n")
+    if (is.null(attr(x, "type"))) FALSE else attr(x, "type") == "single"
```

Comment:
The set of `if` statements inside the conditional section of the main `if` is rather difficult to interpret. 

Although I understand the case of this custom `print`, it feels a bit hacky due to the wide variety of forms the `gbif` object actually can have (`data.frame`, list of `data.frame`s each with varying attributes)... As the main use is to support `print` (polymorphisme), why not create different clearly defined classes `gbif_xxx` each with a easy to write custom print functionality? Adding support for new or adapted `gbif`-like objects would result in far less spaghetti-like code, I think. Still, a matter of choice and maybe not worth the effort to split everything.

Reply:
I had the same feeling; @stijnvanhoey : I started doing it and I realized that it was not really an easy task to make it work for names too. However, the difference in list of dataframes and a single dataframe was alread present. @sckott : what do you think about?

Reply:
i can see the case for separate S3 classes, seems better to open as a separate issue in the main repo